doc/qbk/6.2.MutableString.qbk

```diff
@@ -0,0 +1,71 @@
+[/
+    Copyright (c) 2022 Alan de Freitas (alandefreitas@gmail.com)
+
+    Distributed under the Boost Software License, Version 1.0. (See accompanying
+    file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+    Official repository: https://github.com/CPPAlliance/url
+]
+
+[section:mutablestring MutableString]
+
+A ['MutableString] is a container that has the `assign` and
+`append` member functions that accept iterators to `char`.
+
+The library __decoded_view__ class provides a operations
+to assign and append decoded values to a ['MutableString].
+
+When storing decoded values in a container, these functions
+allow the caller to recycle the input string without
+reallocating memory.
+
+[heading Related Identifiers]
+
+* [link url.ref.boost__urls__decoded_view `decoded_view`]
+* [link url.ref.boost__urls__grammar__is_mutable_string `is_mutable_string`]
+
+[heading Requirements]
+
+In this table:
+
+* `T` is a type meeting the requirements of ['MutableString]
+* `s` is a mutable value of type `T`
+* `first`, `last` are iterators whose `value_type` is `char`
+
+[table Valid expressions
+[[Expression] [Type] [Semantics, Pre/Post-conditions]]
+[
+    [
+    ```
+    s.assign(first,last)
+    ```
+    ]
+    [`T&`]
+    [
+        This member function assigns the `char`s from the
+        range of characters in `[first, last)` to the
+        container `s`.
+    ]
+][
+    [
+    ```
+    s.append(first,last)
+    ```
+    ]
+    [`T&`]
+    [
+        This member function appends the `char`s from the
+        range of characters in `[first, last)` to the
+        container `s`.
```

Comment:
You could say "The character in the range `[ first, last )` are appended to `s`."

But.... this is gonna blow your mind... we _actually_ don't need to say anything, because the postconditions are out of scope for the concept. The only requirement is that it is syntactically valid. It is up to the user to decide what semantics they desire. I'm not sure what the "perfect" way of documenting this is.

Reply:
The column is also about expected Semantics. This is useful even though the user technically able to have different semantics there. The pre- and post-conditions are never part of the concept or are only there as expectations/semantics, because concepts are not able to evaluate that.